devito/passes/clusters/buffering.py

```diff
@@ -177,6 +179,9 @@ def callback(self, clusters, prefix, cache=None):
                 lhs = b.indexed[dims]._subs(dim, b.firstidx.b)
                 rhs = b.function[dims]._subs(dim, b.firstidx.f)
 
+                if dim.is_Conditional:
```

Comment:
what about we place the guard regardless of whether it's a ConditionalDimension or a plain Dimension?

Reply:
second question.

Since `db0` starts, by construction, at 0 and can only take positive values, how about this as a guard instead:

```
guards[pd] = GuardBound(0, b.firstidx.f - b.xd)  # aka: time_M / factor >= 0
```

then later on, you would have to use `andg` to chain this guard with the other `time_M >= time_m` guard

this would generalise to the no-ConditionalDimension case as well

this way, we would generate neater code in my opinion

Reply:
I might be missing something, but I think the guard does need to be on `db0` - we need to check that `db0 + (time_m / factor) - 1 >= 0` or `db0 + (time_M / factor) - 1 >= 0` depending on whether it's forward or backward.

Using `guards[pd] = GuardBound(0, b.firstidx.f - b.xd)  # aka: time_M / factor >= 0` tests fail as expected.